src/main/java/com/gd/clinic/security/jwt/JWTAuthenticationFilter.java

```diff
@@ -0,0 +1,56 @@
+package com.gd.clinic.security.jwt;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.gd.clinic.model.JwtResponseDto;
+import com.gd.clinic.security.service.RefreshTokenService;
+import com.gd.clinic.security.service.UserMain;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.Collections;
+
+public class JWTAuthenticationFilter extends UsernamePasswordAuthenticationFilter {
+
+    @Autowired
+    private final RefreshTokenService refreshTokenService;
+    private final ObjectMapper objectMapper = new ObjectMapper();
+
+    public JWTAuthenticationFilter(RefreshTokenService refreshTokenService) {
+        this.refreshTokenService = refreshTokenService;
+    }
+
+    @Override
+    public Authentication attemptAuthentication(HttpServletRequest request, HttpServletResponse response) throws AuthenticationException {
+        AuthCredentials authCredentials = new AuthCredentials();
+        try {
+            authCredentials = new ObjectMapper().readValue(request.getReader(), AuthCredentials.class);
+        } catch (IOException ignored){
```

Comment:
you should never ignore exceptions. at least for now rethrow `RuntimeException` and pass current one into constructor

Reply:
request.getReader() requires there be an IOException and not a RuntimeException as it currently is in the try-catch

Reply:
I mean that you should not leave the catch block empty, you need to `throw new RuntimeException()` inside